src/test/java/org/reaktivity/reaktor/TestStream.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright 2016-2017 The Reaktivity Project
+ *
+ * The Reaktivity Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package org.reaktivity.reaktor;
+
+import java.util.function.LongSupplier;
+
+import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
+import org.agrona.collections.Long2ObjectHashMap;
+import org.reaktivity.nukleus.function.MessageConsumer;
+import org.reaktivity.nukleus.function.MessagePredicate;
+import org.reaktivity.nukleus.route.RouteManager;
+import org.reaktivity.reaktor.internal.types.control.RouteFW;
+import org.reaktivity.reaktor.internal.types.stream.BeginFW;
+import org.reaktivity.reaktor.internal.types.stream.DataFW;
+import org.reaktivity.reaktor.internal.types.stream.EndFW;
+import org.reaktivity.reaktor.internal.types.stream.FrameFW;
+import org.reaktivity.reaktor.internal.types.stream.ResetFW;
+
+final class TestStream implements MessageConsumer
```

Comment:
Usages of this class should be replaced with mocked calls instead.

Reply:
This is an integration test not a unit test. I could probably rewrite TestStream using Mockito objects but I think the code would be at least as complex, probably more so, and less readable, since much actual behavior is required. Are you suggesting I should be testing in a completely different way, rather than using k3po scripts?

Reply:
done